protocols/virtualfund/virtualfund-single-hop_test.go

```diff
@@ -277,101 +369,89 @@ func TestSingleHopVirtualFund(t *testing.T) {
-				t.Error(err)
-			}
-			if waitingFor != WaitingForCompleteFunding {
-				t.Errorf(`WaitingFor: expected %v, got %v`, WaitingForCompleteFunding, waitingFor)
-			}
-			if o.requestedLedgerUpdates != true {
-				t.Error(`Expected ledger update idempotency flag to be raised, but it wasn't`)
-			}
+			// Cranking should move us to the next waiting point, updating the ledger channel, and alter the extended state to reflect that
+			// TODO: Check that ledger channel is updated as expected
+			oObj, got, waitingFor, _ = o.Crank(&my.privateKey)
 
-			wantRequests := []protocols.GuaranteeRequest{}
+			// Check that the messsages contain the expected ledger proposals
 			switch my.role {
 			case 0:
 				{
-					wantRequests = append(wantRequests, protocols.GuaranteeRequest{
-						ObjectiveId: o.Id(),
-						LedgerId:    ledgerChannelToMyRight.Id,
-						Destination: s.V.Id,
-						Left:        my.destination, Right: p1.destination,
-						LeftAmount:  types.Funds{types.Address{}: big.NewInt(5)},
-						RightAmount: types.Funds{types.Address{}: big.NewInt(5)},
-					})
+					supported, _ := o.ToMyRight.Channel.LatestSupportedState()
+					expectedSignedState := state.NewSignedState(constructLedgerProposal(supported, types.AddressToDestination(alice.address), types.AddressToDestination(p1.address), o.V.Id))
+					_ = expectedSignedState.Sign(&my.privateKey)
+
+					assertSideEffectsContainsMessageWith(got, expectedSignedState, p1, t)
+
 				}
 			case 1:
 				{
-					wantRequests = append(wantRequests, protocols.GuaranteeRequest{
-						ObjectiveId: o.Id(),
-						LedgerId:    ledgerChannelToMyLeft.Id,
-						Destination: s.V.Id,
-						Left:        alice.destination, Right: my.destination,
-						LeftAmount:  types.Funds{types.Address{}: big.NewInt(5)},
-						RightAmount: types.Funds{types.Address{}: big.NewInt(5)},
-					})
-					wantRequests = append(wantRequests, protocols.GuaranteeRequest{
-						ObjectiveId: o.Id(),
-						LedgerId:    ledgerChannelToMyRight.Id,
-						Destination: s.V.Id,
-						Left:        my.destination, Right: bob.destination,
-						LeftAmount:  types.Funds{types.Address{}: big.NewInt(5)},
-						RightAmount: types.Funds{types.Address{}: big.NewInt(5)},
-					})
-				}
-			case 2:
-				{
-					wantRequests = append(wantRequests, protocols.GuaranteeRequest{
-						ObjectiveId: o.Id(),
-						LedgerId:    ledgerChannelToMyLeft.Id,
-						Destination: s.V.Id,
-						Left:        p1.destination, Right: my.destination,
-						LeftAmount:  types.Funds{types.Address{}: big.NewInt(5)},
-						RightAmount: types.Funds{types.Address{}: big.NewInt(5)},
-					})
+					supported, _ := o.ToMyRight.Channel.LatestSupportedState()
+					expectedSignedState := state.NewSignedState(constructLedgerProposal(supported, types.AddressToDestination(p1.address), types.AddressToDestination(bob.address), o.V.Id))
+					_ = expectedSignedState.Sign(&my.privateKey)
+
+					assertSideEffectsContainsMessageWith(got, expectedSignedState, bob, t)
```

Comment:
Check for understanding:

In the new version, case 0 and case 1 both seem to be checking ledger channels to the right, and case 2 is eliminated. This is because the only checks performed here are in channels where `my` is the leader.

Reply:
Yup, that's correct. We are only expecting a proposal in the scenarios where `my` is the leader. I'll add some comments so that is clear.